src/riscv.c

```
@@ -357,7 +370,10 @@ static void load_dtb(char **ram_loc, vm_attr_t *attr)
             const char *name = fdt_get_name(dtb_buf, subnode, NULL);
             assert(name);
 
-            uint32_t addr = strtoul(name + 7, NULL, 16);
+            char *at_pos = strchr(name, '@');
+            assert(at_pos);
+
+            uint32_t addr = strtoul(at_pos + 1, NULL, 16);
```

No endptr validation for strtoul. Malformed DTB could cause silent errors.

Consider:
```c
  char *endptr;
  uint32_t addr = strtoul(at_pos + 1, &endptr, 16);
  if (endptr == at_pos + 1) {
      rv_log_error("Invalid unit-address in node: %s", name);
      // handle error
  }
```